### DISCUSSION TITLE
Get lang from url param

### DISCUSSION THREAD

**Discussion by sc00** [sveltekit-i18n/lib#47] [Q&A] [answered]
Hi,

I haven't found any info on handling localization through url params. E.g.:

mydomain.com/?lang=de

What would be a good practice to realize this with SSR? Is this what locale.get() is for? Or would you want to set the defaultLocale to something like searchParams.get('lang')?

Thanks and cheers

**Comment by jarda-svoboda** [comment]
hi.)

i would use the same mechanism as described [here](https://github.com/sveltekit-i18n/lib/tree/master/examples/locale-router), but you would need to implement (probably) cookie storage for `locale` persistence. the cookie `locale` should be used within `__layout.svelte` and `hooks.js` files in case there is no url `lang` param. also `[lang]` folder in `routes` should be removed.

`locale.get()` is just a shortcut for Svelte's `get` method with `locale` as a param – it's useful in case you are using your locale within `.js` files so there is no need for importing the `get` over and over...

**Comment by jarda-svoboda** [comment, marked as answer]
Here is an example for current version: https://github.com/sveltekit-i18n/lib/blob/master/examples/locale-param 
You can preview it [here](https://locale-param.netlify.app)

[a comment by an automated account is not shown]
